dali/core/mm/async_pool_test.cu

```diff
@@ -351,5 +350,103 @@ TEST(MMAsyncPool, CrossStreamWithHogs) {
   upstream.check_leaks();
 }
 
+#if DALI_USE_CUDA_VM_MAP
+
+TEST(MM_VMAsyncPool, MultiThreadedSingleStreamRandom) {
+  if (!cuvm::IsSupported())
+    GTEST_SKIP() << "Virtual memory management API is not supported on this machine.";
+
+  CUDAStream stream = CUDAStream::Create(true);
+  {
+    vector<block> blocks;
+    std::mutex mtx;
+
+    async_pool_resource<memory_kind::device, cuda_vm_resource> pool;
+
+    vector<std::thread> threads;
+
+    for (int t = 0; t < 10; t++) {
+      threads.push_back(std::thread([&]() {
+        AsyncPoolTest(pool, blocks, mtx, stream);
+      }));
+    }
+    for (auto &t : threads)
+      t.join();
+  }
+}
+
+TEST(MM_VMAsyncPool, MultiThreadedMultiStreamRandom) {
+  if (!cuvm::IsSupported())
+    GTEST_SKIP() << "Virtual memory management API is not supported on this machine.";
+
+  async_pool_resource<memory_kind::device, cuda_vm_resource> pool;
+
+  vector<std::thread> threads;
+
+  for (int t = 0; t < 10; t++) {
+    threads.push_back(std::thread([&]() {
+      CUDAStream stream = CUDAStream::Create(true);
+      vector<block> blocks;
+      detail::dummy_lock mtx;
+      AsyncPoolTest(pool, blocks, mtx, stream);
+      CUDA_CALL(cudaStreamSynchronize(stream));
+    }));
+  }
+  for (auto &t : threads)
+    t.join();
```

Comment:
Is it possible to check leaks?

Reply:
Not really. If there was an access to the global pool itself (but it's private) I could check the data returned by `get_stat().`
That said, `async_resource` behaves the same way with both pool types, so if there are no leaks with default pool, there would be no leaks from `cuda_vm_resource`.